### PR TITLE
ROC-7438 tech debt: intellij inspections

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-7438

### Change description ###
Timeboxed exercise going through IntelliJ inspection errors and warnings.
Each type of inspection is dealt with in individual commits so reviewers are advised to go through the PR by commit rather than as a whole.

Feel free to disagree with entire commits (inspections) or individual changes. I'll only keep the ones that have zero contention.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
Note: I was trying to use the `!fixup` `--autosquash` process to keep this clean, but it looks like all of the amendments I've made have gone into one final commit instead of the intended ones. Must have done something wrong :(

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/evidence/CCDEvidenceType.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
     STATEMENT_OF_ACCOUNT("Statements of account"),
     OTHER("Other");
 
-    String description;
+    final String description;
```
Comment (daveoh) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/evidence/CCDEvidenceType.java]: Access should be private to be consistent.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java [hunk 2]
```diff
@@ -16,19 +16,17 @@ public void to(CCDRespondent.CCDRespondentBuilder builder, Claim claim) {
...
 
         claim.getReDeterminationRequestedAt().ifPresent(builder::redeterminationRequestedDate);
     }
 
     public void from(Claim.ClaimBuilder builder, CCDRespondent respondent) {
         if (StringUtils.isBlank(respondent.getRedeterminationExplanation())
-            && respondent.getRedeterminationMadeBy() == null
-            && respondent.getRedeterminationRequestedDate() == null
+            || respondent.getRedeterminationMadeBy() == null
+            || respondent.getRedeterminationRequestedDate() == null
```
Comment (daveoh) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java]: Why was this change made? Is there a situation where only one of these could be blank/null?

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/InterestMapperTest.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 public class InterestMapperTest {
 
-    InterestMapper interestMapper = new InterestMapper(new InterestBreakdownMapper(new MoneyMapper()),
+    final InterestMapper interestMapper = new InterestMapper(new InterestBreakdownMapper(new MoneyMapper()),
```
Comment (daveoh) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/InterestMapperTest.java]: Does this variable need default access?

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java [hunk 4]
```diff
@@ -357,7 +362,8 @@ public void mapFromCCDDefendantWithSettlements() {
         // Then
         assertNotNull(finalClaim.getSettlementReachedAt());
         assertNotNull(finalClaim.getSettlement());
-        assertThat(finalClaim.getSettlement().get().getPartyStatements().size(), is(3));
+        assertThat(finalClaim.getSettlement().map(Settlement::getPartyStatements).map(List::size).orElse(0),
```
Comment (daveoh) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: This feels like an overcomplicated way of doing this.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/claimantresponse/ClaimantResponseType.java [hunk 5]
```diff
@@ -7,7 +7,7 @@
     ACCEPTATION("acceptation"),
     REJECTION("rejection");
 
-    String description;
+    final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/claimantresponse/ClaimantResponseType.java]: does the access need to be default?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/claimantresponse/FormaliseOption.java [hunk 6]
```diff
@@ -8,7 +8,7 @@
     SETTLEMENT("Settlement"),
     REFER_TO_JUDGE("Refer to Judge");
 
-    String description;
+    final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/claimantresponse/FormaliseOption.java]: does the access need to be default?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java [hunk 7]
```diff
@@ -153,10 +153,10 @@ private boolean lastStatementIsOffer() {
         return getLastStatement().getType().equals(StatementType.OFFER);
     }
 
-    private boolean lastStatementIsAcceptationNotBy(MadeBy madeBy) {
+    private boolean lastStatementIsNotAcceptationBy(MadeBy madeBy) {
         PartyStatement lastStatement = getLastStatement();
-        return lastStatement.getType().equals(StatementType.ACCEPTATION)
-            && !lastStatement.getMadeBy().equals(madeBy);
+        return !lastStatement.getType().equals(StatementType.ACCEPTATION)
+            || lastStatement.getMadeBy().equals(madeBy);
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java]: I feel this was easier to understand before this change.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/response/ResponseType.java [hunk 8]
```diff
@@ -5,7 +5,7 @@
     FULL_ADMISSION("I admit all of the claim"),
     PART_ADMISSION("I admit part of the claim");
 
-    String description;
+    final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/response/ResponseType.java]: does the access need to be default?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/BankAccount.java [hunk 9]
```diff
@@ -20,7 +20,7 @@
         ISA("ISA"),
         OTHER("Other");
 
-        String description;
+        final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/BankAccount.java]: does the access need to be default?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/Child.java [hunk 10]
```diff
@@ -19,7 +19,7 @@
         BETWEEN_11_AND_15("Between 11 and 15"),
         BETWEEN_16_AND_19("Between 16 and 19");
 
-        String description;
+        final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/Child.java]: does the access need to be default?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/DisabilityStatus.java [hunk 11]
```diff
@@ -5,7 +5,7 @@
     YES("Yes"),
     SEVERE("Severe");
 
-    String description;
+    final String description;
```
Comment (daveoh) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/DisabilityStatus.java]: does the access need to be default? 
.... same for other beans in this package

File: job-scheduler/src/main/java/uk/gov/hmcts/cmc/scheduler/services/JobService.java [hunk 12]
```diff
@@ -28,7 +28,7 @@
 @Service
 public class JobService {
 
-    Logger logger = LoggerFactory.getLogger(this.getClass());
+    final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (daveoh) [inline job-scheduler/src/main/java/uk/gov/hmcts/cmc/scheduler/services/JobService.java]: does the access need to be default?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/IntentionToProceedService.java [hunk 13]
```diff
@@ -32,7 +32,7 @@
 @Service
 public class IntentionToProceedService {
 
-    Logger logger = LoggerFactory.getLogger(this.getClass());
+    final Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (daveoh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/IntentionToProceedService.java]: does the access need to be default?

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java (hunk 4 above)
Comment (daveoh) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: Maybe : 
```suggestion
assertThat(finalClaim.getSettlement().map( s -> s.getPartyStatements().size()).orElse(0), is(3))
```
Thoughts?

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java (hunk 2 above)
Comment (andrewwa-kainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java]: If there is, then we're now protected against NPEs and if there isn't then it's made no difference.

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java (hunk 4 above)
Comment (andrewwa-kainos) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: Will revert and add a suppression

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java (hunk 7 above)
Comment (andrewwa-kainos) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/offers/Settlement.java]: Will revert and add a suppression

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  -2
+ src/main/java/uk/gov/hmcts/cmc/claimstore/utils/ResponseHelper.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4890211&bid=16196854)

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/models/GeneratePinRequest.java [hunk 14]
```diff
@@ -7,7 +7,7 @@
 public class GeneratePinRequest {
 
     public final String firstName;
-    @JsonInclude(JsonInclude.Include.ALWAYS)
+    @JsonInclude()
```
Comment (ckalista) [inline ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/models/GeneratePinRequest.java]: ALWAYS by default 👍

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java (hunk 2 above)
Comment (ckalista) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ReDeterminationMapper.java]: Good spot @daveoh. I think I am happy this this explanation.

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/DataFixHandler.java [hunk 15]
```diff
@@ -174,35 +174,24 @@ private void logMessageForCsv(
...
-            .append(",")
-            .append(lastEventDetails.getCreatedDate())
-            .append(",")
-            .append(lastEventDetails.getUserFirstName())
-            .append(" ")
-            .append(lastEventDetails.getUserLastName())
-            .append(",")
-            .append(hasProgressed)
-            .toString()
+        logger.info("CSV,"
```
Comment (ckalista) [inline ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/DataFixHandler.java]: I am surprised we still have this {{ccd-claim-migration}}, really we should get rid of this but for now this is ok.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/amount/Amount.java [hunk 16]
```diff
@@ -3,7 +3,7 @@
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
```
Comment (ckalista) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/amount/Amount.java]: JsonTypeInfo.As.PROPERTY is also default 👍

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/AmountContentProvider.java [hunk 17]
```diff
@@ -78,17 +78,16 @@ public AmountContent create(Claim claim) {
             interestRealValue = interestContent.getAmountRealValue();
         }
 
+        BigDecimal subtotalAmount = ccjAmount
```
Comment (ckalista) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/AmountContentProvider.java]: 👍

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/PhoneNumberConstraintValidator.java [hunk 18]
```diff
@@ -24,7 +24,7 @@ private boolean isWithInLengthLimit(String normalized) {
 
     private String normalize(String value) {
         return value
-            .replaceAll("\\(|\\)| |-|\\+", "")
+            .replaceAll("[() \\-+]", "")
```
Comment (ckalista) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/PhoneNumberConstraintValidator.java]: Can we just get a tester to cast an eye on this?

Comment (andrewwa-kainos) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/PhoneNumberConstraintValidator.java]: Added a note on the JIRA ticket

**Pull request merged**